sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -118,7 +118,7 @@ class SimpleFunctionRegistry extends FunctionRegistry with Logging {
         throw new AnalysisException(s"undefined function $name")
       }
     }
-    func(children)
+    func(children).setFuncName(name.funcName)
```

Comment:
Hm, do functions not already have names somewhere to use, that can already be set differently per alias? it looks like that's what nodeName is for, and it's already overridden in the aliases, so I'm missing why this is different.

Reply:
> do functions not already have names somewhere to use, that can already be set differently per alias?

I think, No? Currently, when we use alias of bool_and (i.e, every), it will be resolved as a constructor of `BoolAnd` using `FunctionRegistry#expressions`, which sets `bool_and' as a nodeName.
https://github.com/apache/spark/blob/9351e3e76fb11e9fdaf39aef5aea86fdeccd6f28/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/UnevaluableAggs.scala#L55-L57

Reply:
This may not make sense, I don't know this code well, but: is it not that "Exists" needs to customize its nodeName, for example? or does it never exist as such a node.

Reply:
`every` here does not have any node. It will be resolved as a `BoolAnd`.
https://github.com/apache/spark/blob/32af7004a254107340df908e8dd91a0bc22e068e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L316-L320

Reply:
I see. I don't know enough to evaluate the effect of changing nodeName for all implementations, which seems like a broader change than required, but it does make some sense. Maybe @liancheng or @yhuai has an opinion.